### ISSUE TITLE
Disable AliasUsage Checks in `@spec`

### ISSUE THREAD

**Issue by kennyballou** [rrrene/credo#133]
Correct me if I'm wrong, but my understanding of dialyzer is that it does not understand Elixir's aliasing, therefore, `@spec` lines should not be expected to use aliases.

I've created a sample test case that I was planning to use for a pull request, but I don't yet have the time for solving how to ignore aliases in specs:

```elixir
  test "it should NOT report violation in `@spec`s" do
"""
defmodule Sample do
  @spec foo(Sample.Foo.t) :: false
  def foo(%Sample.Foo{} = _), do: false
end
""" |> to_source_file
    |> refute_issues(@described_check)
  end
```

I believe where I'm fumbling with this problem is that by the time we have found an alias, we are too far into the tree to backtrack to know if it's inside a `:spec` tuple[1].

[1]: https://github.com/rrrene/credo/blob/master/lib/credo/check/design/alias_usage.ex#L118

**Comment by rrrene** [comment]
:+1: Thx for reporting!

**Comment by rrrene** [comment]
`v0.4.7` contains the fix. Thx again!

**Comment by kennyballou** [comment]
Thanks for the quick fix! I'll let you know if I see anything else. :+1:
